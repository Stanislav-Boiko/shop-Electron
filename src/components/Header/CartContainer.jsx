

import React from 'react';
import { Link } from "react-router-dom";
import Cart from '../../assets/Cart.svg'
import Liked from '../../assets/Liked.svg'

import './CartContainer.scss'

function CartContainer({cartLength, favoritesLength}) {

    
    
    return(
         
        <div  className='cartContainer'>
            
            <Link to='/favorites' className='cart'>
                <span className="counter">{favoritesLength}</span>
                <img src={Liked} className='CartImg' alt="Liked" />
            </Link>

            
            <Link to='/cart' className='cart'>
                <span className="counter">{cartLength}</span> 
                <img src={Cart} className='CartImg' alt="Cart" />
            </Link>

        </div>
  
    )
}

export default CartContainer;