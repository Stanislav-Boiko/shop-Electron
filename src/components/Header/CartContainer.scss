
.cartContainer {
    display: flex;
    gap: 12px;
    margin-left: 40px;
    
    
    .cart {
        display: flex;
        align-items: center;
        border-radius: 8px;
        border: 1px solid #8a33fd;
        background-color: #fff;
        gap: 12px;
        padding: 10px;
        text-decoration: none;
        cursor: pointer;

        &:hover {
            border: 1px solid #6a0de3;

            .counter,.CartImg {
                transform: scale(1.1);
            }
        }

        .counter {
            color: #6a0de3;
            font-size: 18px;
            
        }

        .CartImg {
            filter: invert(35%) sepia(93%) saturate(7500%) hue-rotate(255deg) brightness(95%) contrast(105%);
        }
    }
}