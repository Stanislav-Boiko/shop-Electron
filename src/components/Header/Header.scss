.header {
    display: block;
    height: 78px;
    width: 100%;
    background-color: #fff;
    border-bottom: 3px solid #8a33fd;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 1;

    .nav {
        width: 100%;
        max-width: 1240px; // <-- так само, як .container
        margin: 0 auto;
        display: flex;
        justify-content: space-between;
        padding: 10px 32px; // Додаємо базовий padding
        align-items: center;
        gap: 10px;
        box-sizing: border-box;

        &_logo {
            font-family: Arial, Helvetica, sans-serif;
            font-size: 50px;
            font-weight: 600;
            color: #6a0de4;
            text-decoration: none;
            margin-right: 10px;
        }

        &_item {
            font-size: 24px;
            font-weight: 400;
            text-decoration: none;
            color: #8a33fd;
            padding: 3px;
            margin-right: 10px;

            &:hover {
                transform: scale(1.1);
            }
        }

        .active {
            font-weight: 600;
            color: #6a0de4;
            text-decoration: underline;
        }

        @media (max-width: 1300px) {
            padding: 10px 20px;
        }
        @media (max-width: 900px) {
            padding: 10px 10px;
        }
        @media (max-width: 600px) {
            padding: 10px 4vw;
        }
    }
}
