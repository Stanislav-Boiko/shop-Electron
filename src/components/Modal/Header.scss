


.titleModal {
    margin: 0 0 32px 0;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 32px;
    font-weight: 400;
    line-height: 38.73px;
    color: #6a0de4;
  }