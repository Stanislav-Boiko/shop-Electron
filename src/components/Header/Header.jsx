import React from 'react';
import { Link, useLocation } from "react-router-dom";
import CartContainer from "./CartContainer";
import './Header.scss'

function Header({cartLength, favoritesLength}) {
    const location = useLocation(); // Отримуємо поточний шлях
    const currentPath = location.pathname;

    return (
        <header className="header">
               
            <div className='nav'>
                <Link
                    to='/shop'
                    className='nav_logo'
                >
                    Electron
                </Link>

                <Link
                    to='/shop'
                    className={`nav_item ${currentPath === '/shop' ? 'active' : ''}`}
                >
                    Shop
                </Link>

                <Link
                    to='/about'
                    className={`nav_item ${currentPath === '/about' ? 'active' : ''}`}
                >
                    About
                </Link>

                <Link
                    to='/services'
                    className={`nav_item ${currentPath === '/services' ? 'active' : ''}`}
                >
                    Services
                </Link>

                <Link
                    to='/contacts'
                    className={`nav_item ${currentPath === '/contacts' ? 'active' : ''}`}
                >
                    Contacts
                </Link>

                <CartContainer cartLength={cartLength} favoritesLength={favoritesLength} />
            </div>
            
        </header>
    );
}

export default Header;
