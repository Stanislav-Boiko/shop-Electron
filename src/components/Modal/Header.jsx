import React from 'react';
import './Header.scss';


function Header ({ children }) {
    return (
        
        <h2 className="titleModal">{children}</h2>

    )
};


export default Header;