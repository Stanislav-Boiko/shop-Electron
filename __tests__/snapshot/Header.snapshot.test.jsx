import React from 'react';
import { render } from '@testing-library/react';

import Header from 'components/Header/Header';


import { MemoryRouter } from 'react-router-dom';


test('Header рендериться правильно', () => {
    // Обгортаємо Header у MemoryRouter, бо він використовує useLocation та Link
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/shop']}>
        <Header cartLength={2} favoritesLength={3} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  })