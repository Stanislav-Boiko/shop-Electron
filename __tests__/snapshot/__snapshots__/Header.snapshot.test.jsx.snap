// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header рендериться правильно 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="nav"
    >
      <a
        class="nav_logo"
        data-discover="true"
        href="/shop"
      >
        Electron
      </a>
      <a
        class="nav_item active"
        data-discover="true"
        href="/shop"
      >
        Shop
      </a>
      <a
        class="nav_item "
        data-discover="true"
        href="/about"
      >
        About
      </a>
      <a
        class="nav_item "
        data-discover="true"
        href="/services"
      >
        Services
      </a>
      <a
        class="nav_item "
        data-discover="true"
        href="/contacts"
      >
        Contacts
      </a>
      <div
        class="cartContainer"
      >
        <a
          class="cart"
          data-discover="true"
          href="/favorites"
        >
          <span
            class="counter"
          >
            3
          </span>
          <img
            alt="Liked"
            class="CartImg"
            src="test-file-stub"
          />
        </a>
        <a
          class="cart"
          data-discover="true"
          href="/cart"
        >
          <span
            class="counter"
          >
            2
          </span>
          <img
            alt="Cart"
            class="CartImg"
            src="test-file-stub"
          />
        </a>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
